test/Microsoft.Extensions.Diagnostics.HealthChecks.Tests/DefaultHealthCheckServiceTest.cs

```diff
@@ -188,6 +188,35 @@ public async Task CheckHealthAsync_SetsRegistrationForEachCheck()
                 });
         }
 
+        [Fact]
+        public async Task CheckHealthAsync_Cancellation_CanPropagate()
+        {
+            // Arrange
+            var insideCheck = new TaskCompletionSource<object>();
+
+            var service = CreateHealthChecksService(b =>
+            {
+                b.AddAsyncCheck("cancels", async ct =>
+                {
+                    insideCheck.SetResult(null);
+
+                    await Task.Delay(10000, ct);
```

Comment:
Could you use a `ManualResetEvent` here?

Reply:
I could do a lot of things, why would I want to do that?

Reply:
Discussed offline, the delay doesn't actually happen because cancellation happens right away